illumina_uploader/illumina_uploader.py

```diff
@@ -119,65 +122,99 @@ def main():
             logger.info("This process will exit now, and {0} will get picked up by the other illumina-uploader process".format(single_run))
             exit(0)
         else:
+            sleeptimeInSeconds = int(localInfo["sleeptime"])*60
+            ploverEmailArgs = {
+                "debug": isDebug,
+                "mailto": ploverEmailInfo["mailto"],
+                "token": ploverEmailInfo["emailtoken"],
+                "emailurl": ploverEmailInfo["emailurl"],
+            }
+            folderToUpload = ""
+            reason = ""
+            shouldStopEmails = False
+            if mcmsEmailInfo['enabled']:
+                if not(field in mcmsEmailInfo for field in ['authurl','emailurl','clientid','clientsecret','senderemail','recipientemail']):
+                    logger.error('MCMS incorrectly configured')
+                    mcmsEmailInfo['enabled'] = False
+                else:
+                    mcms = McmsEmailService(mcmsEmailInfo['authurl'], mcmsEmailInfo['clientid'], mcmsEmailInfo['clientsecret'], mcmsEmailInfo['senderemail'], logger)
+
             #Call fabric tasks
             while(True):
                 logger.info("Start Watching Directores: {0}".format(",".join(inputDirs)))
                 #runsCache stores run info for later retrieval. TODO optimize
                 runsCache = None
                 #Default mail args
-                mailArgs = {
-                    "debug": isDebug,
-                    "token": emailInfo["emailtoken"],
-                    "mailto": emailInfo["mailto"]
-                }
                 try:
                     runsCache = dbObject.watchDirectories(localInfo["watchfilepath"], inOutMap)
-                except KeyboardInterrupt as error:
```

Comment:
This never made sense. This is only for physical intervention which probably never happens unless someone just wants to stop the program and not just to stop this current sync of folders.